### PR TITLE
chore(ckerc20): NNS proposal to upgrade ledger suite orchestrator

### PR DESCRIPTION
**Pull request by gregorydemay**
(XC-148): Proposal to upgrade the ledger suite orchestrator and all managed ledger suites to the latest release candidate `rc--2024-07-18_01-30--github` (`de29a1a55b589428d173b31cdb8cec0923245657`)

### CODE REVIEW

File: rs/ethereum/cketh/mainnet/orchestrator_upgrade_2024_07_22.md [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+Upgrading the corresponding index and archive canisters of all managed ledger suites, while not strictly necessary, simplifies version management by ensuring that all canisters use the code from the same git commit.
+
+
+## Upgrade args
+
+```
+git fetch
+git checkout de29a1a55b589428d173b31cdb8cec0923245657
+cd rs/ethereum/ledger-suite-orchestrator
+didc encode -d ledger_suite_orchestrator.did -t '(OrchestratorArg)' '(variant { UpgradeArg = record {git_commit_hash = opt "de29a1a55b589428d173b31cdb8cec0923245657"; ledger_compressed_wasm_hash = opt "9495d67c6e9ab4cec3740d68fa0a103dbcfd788d978f4cb58308d847d59e635b"; index_compressed_wasm_hash = opt "12612d5b2565b018a01fa5007c1de54c750c74dbd485f1ef31d71d689177fa86"; archive_compressed_wasm_hash = opt "080d7f03f18ca1e1804626b2cefe06f732a01f64f7d77e1a2c6bfea5a4b4baa1"}})'
```
Comment (gregorydemay) [inline rs/ethereum/cketh/mainnet/orchestrator_upgrade_2024_07_22.md:33]: `4449444c0e6b03ba9edbce040186dac7c60906c3cab9d20a0b6c05b59798d00202e898cad20405e9c2d6a70905a1eedde50a0598c6c2cf0b056e036c04e7aba21404cdd1f4e20404e4d8cce80b0494e2c9d00c046e7d6e716c03b59798d00207ffb08aab0809cb83d3c90f0a6e086c04e7aba2147dcdd1f4e2047de4d8cce80b7d94e2c9d00c7d6d686e686c0292bec7bd030cfdb5c9a9070d6c0299e7a4cf087db4e3ade809716c05c295a993017b9efeb9a40371f2c794ae037db1a38bf40d7191c9aafe0d7101000000014030383064376630336631386361316531383034363236623263656665303666373332613031663634663764373765316132633662666561356134623462616131012864653239613161353562353839343238643137336233316364623863656330393233323435363537014039343935643637633665396162346365633337343064363866613061313033646263666437383864393738663463623538333038643834376435396536333562014031323631326435623235363562303138613031666135303037633164653534633735306337346462643438356631656633316437316436383931373766613836`

File: rs/ethereum/cketh/mainnet/orchestrator_upgrade_2024_07_22.md [hunk 2]
```diff
@@ -0,0 +1,140 @@
+# Proposal to upgrade the ledger suite orchestrator canister
```
Comment (gregorydemay) [inline rs/ethereum/cketh/mainnet/orchestrator_upgrade_2024_07_22.md:1]:
```
~/bin/ic-admin \
    --use-hsm \
    --key-id 01 \
    --slot 0 \
    --pin $(cat ~/.hsm-pin) \
    --nns-url "https://ic0.app" \
    propose-to-change-nns-canister \
    --proposer 17212304975669116357 \
    --canister-id vxkom-oyaaa-aaaar-qafda-cai \
    --mode upgrade \
    --wasm-module-path ./ic-ledger-suite-orchestrator-canister.wasm.gz \
    --wasm-module-sha256 81f426bcc52140fdcf045d02d00b04bfb4965445b8aed7090d174fcdebf8beea \
    --arg args.bin \
    --summary-file ./orchestrator_upgrade_2024_07_22.md
```

File: rs/ethereum/cketh/mainnet/orchestrator_upgrade_2024_07_22.md (hunk 1 above)
Comment (manudrijvers) [inline rs/ethereum/cketh/mainnet/orchestrator_upgrade_2024_07_22.md:33]: ```4449444c0e6b03ba9edbce040186dac7c60906c3cab9d20a0b6c05b59798d00202e898cad20405e9c2d6a70905a1eedde50a0598c6c2cf0b056e036c04e7aba21404cdd1f4e20404e4d8cce80b0494e2c9d00c046e7d6e716c03b59798d00207ffb08aab0809cb83d3c90f0a6e086c04e7aba2147dcdd1f4e2047de4d8cce80b7d94e2c9d00c7d6d686e686c0292bec7bd030cfdb5c9a9070d6c0299e7a4cf087db4e3ade809716c05c295a993017b9efeb9a40371f2c794ae037db1a38bf40d7191c9aafe0d7101000000014030383064376630336631386361316531383034363236623263656665303666373332613031663634663764373765316132633662666561356134623462616131012864653239613161353562353839343238643137336233316364623863656330393233323435363537014039343935643637633665396162346365633337343064363866613061313033646263666437383864393738663463623538333038643834376435396536333562014031323631326435623235363562303138613031666135303037633164653534633735306337346462643438356631656633316437316436383931373766613836```

**Review by manudrijvers — APPROVED** [review]

File: rs/ethereum/cketh/mainnet/orchestrator_upgrade_2024_07_22.md (hunk 1 above)
Comment (gregorydemay) [reply rs/ethereum/cketh/mainnet/orchestrator_upgrade_2024_07_22.md:33]: confirmed ([here](https://github.com/dfinity/ic/pull/427#discussion_r1682485056))

File: rs/ethereum/cketh/mainnet/orchestrator_upgrade_2024_07_22.md [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+Previous ledger suite orchestrator proposal: https://dashboard.internetcomputer.org/proposal/131053
+
+---
+
+## Motivation
+
+This upgrades the ledger suite orchestrator and all managed ledger suites to the latest version.
+
+The motivation to upgrade the orchestrator are mainly the following points:
+1. Add logic to upgrade managed canisters.
```
Comment (THLO) [inline rs/ethereum/cketh/mainnet/orchestrator_upgrade_2024_07_22.md:18]: I think this sentence is a bit (too) ambiguous. Is logic added to make upgrades possible or is the existing upgrade logic extended?

Comment (gregorydemay) [reply rs/ethereum/cketh/mainnet/orchestrator_upgrade_2024_07_22.md:18]: When we released it, we didn't have yet the logic to upgrade managed canisters. So how about
```suggestion
1. Add logic to make upgrade of managed canisters possible.
```

File: rs/ethereum/cketh/mainnet/orchestrator_upgrade_2024_07_22.md (hunk 1 above)
Comment (THLO) [inline rs/ethereum/cketh/mainnet/orchestrator_upgrade_2024_07_22.md:33]: All hashes look good.  :+1:

**Review by THLO — APPROVED** [review]

**Pull request merged**